src/pages/studyView/StudyViewPage.tsx

```diff
@@ -12,63 +12,64 @@ import { PieChart } from './charts/pieChart/PieChart'
 import { ClinicalAttribute } from 'shared/api/generated/CBioPortalAPI';
 import styles from "./styles.module.scss";
 
-
+export type ClinicalDataType= "SAMPLE" | "PATIENT";
 export type ClinicalAttributeData = {[attrId:string]:ClinicalDataCount[]};
-export type ClinicalAttributeDataWithMeta = {attributeId:string,clinicalDataType:"SAMPLE" | "PATIENT",counts:ClinicalDataCount[]};
+export type ClinicalAttributeDataWithMeta = {attributeId:string,clinicalDataType:ClinicalDataType,counts:ClinicalDataCount[]};
 export class StudyViewPageStore {
 
     constructor(){
 
     }
 
-    @observable studyId = "hnsc_tcga"
+    @observable studyId = "hnsc_tcga";
 
-    @observable sampleAttrIds = ["OCT_EMBEDDED"]
+    @observable sampleAttrIds = ["OCT_EMBEDDED"];
 
-    @observable patientAttrIds = ["AJCC_PATHOLOGIC_TUMOR_STAGE","ICD_O_3_HISTOLOGY"]
-    
-    @observable filter:StudyViewFilter;
+    @observable patientAttrIds = ["AJCC_PATHOLOGIC_TUMOR_STAGE","ICD_O_3_HISTOLOGY"];
 
-    @observable propFilters:{[id:string]:ClinicalDataEqualityFilter} = {}
+    @observable.ref clinicalDataEqualityFilter:{[id:string]:ClinicalDataEqualityFilter} = {};
 
-    @action updateFilters( attributeId:string,
-        clinicalDataType: "SAMPLE" | "PATIENT",
-                           value:string) {
+    @action updateClinicalDataEqualityFilters( attributeId      : string,
+                                               clinicalDataType : ClinicalDataType,
+                                               value            : string) {
         
-        let id = [clinicalDataType,attributeId].join('_')
+        let id = [clinicalDataType,attributeId].join('_');
 
-        let filters: StudyViewFilter = this.filter||{}
+        let clinicalDataEqualityFilter = _.merge({},this.clinicalDataEqualityFilter)
```

Comment:
you don't need to clone this thing.  it's the item that is being edited that should be cloned rather than edited.  e.g. this.clinicalDataEqualityFilter[id]